test/e2e/metrics_test.go

```diff
@@ -120,3 +132,107 @@ func checkTargetNodes(windowsEndpoints *core.Endpoints) error {
 
 	return nil
 }
+
+// PrometheusQuery defines the result of the /query request
+// Example Reference of Prometheus Query Response: https://prometheus.io/docs/prometheus/latest/querying/api/
+type PrometheusQuery struct {
+	Status string `json:"status"`
+	Data   Data   `json:"data"`
+}
+
+// Data defines the response content of the prometheus server to the given request
+type Data struct {
+	Result []Result `json:"result"`
+}
+
+// Result specifies information about the metric in the query and the resulting value
+type Result struct {
+	Metric Metric        `json:"metric"`
+	Value  []interface{} `json:"value"`
+}
+
+// Metric helds information regarding the metric defined in the query
+type Metric struct {
+	Instance string `json:"instance"`
+}
+
+// testWindowsPrometheusRules tests if prometheus rules specific to Windows are defined by WMCO
+// It also tests the Prometheus queries by sending http requests to Prometheus server.
+func testWindowsPrometheusRules(t *testing.T) {
+	err := framework.AddToFrameworkScheme(monitoringv1.AddToScheme, &monitoringv1.PrometheusRule{})
+	require.NoError(t, err)
+	err = framework.AddToFrameworkScheme(route.AddToScheme, &route.Route{})
+	require.NoError(t, err)
+
+	// test if PrometheusRule object exists in WMCO repo
+	promRule := &monitoringv1.PrometheusRule{}
+	err = framework.Global.Client.Get(context.TODO(),
+		types.NamespacedName{Namespace: "openshift-windows-machine-config-operator",
+			Name: prometheusRule}, promRule)
+	require.NoError(t, err)
+	// test if rules specific to windows exist
+	require.Equal(t, windowsRuleName, promRule.Spec.Groups[0].Name)
+
+	// get route to access Prometheus server instance in monitoring namespace
+	prometheusRoute := &route.Route{}
+	err = framework.Global.Client.Get(context.TODO(),
+		types.NamespacedName{Namespace: monitoringNamespace, Name: "prometheus-k8s"}, prometheusRoute)
+	require.NoError(t, err)
+
+	// define authorization token required to call Prometheus api
+	var bearer = "Bearer " + gc.prometheusToken
+	// InsecureSkipVerify is required to avoid errors due to bad certificate
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	// Following tests carry out api calls to Prometheus server with all the queries defined in the Windows PrometheusRule
+	// object. The tests check if the query results have instances corresponding to configured Windows Nodes.
+	// They also test if the metrics returned have a non-zero value.
+	for _, rules := range promRule.Spec.Groups {
+		if rules.Name == windowsRuleName {
```

Comment:
One way to reduce indentation is to:
```suggestion
if rules.Name != windowsRuleName {
    continue
}
//rest of the code with less indentation
```

Reply:
Now for whatever reason there is no Windows rules present, it will result in a false positive. So you need to check for that also.

Reply:
We are testing for existence of `windows.rules` [here](https://github.com/openshift/windows-machine-config-operator/pull/234/commits/d541bb524846b3069608786d52617fbd7ff8a8eb#diff-41354c8d32cc5449ea0e5de5e5f28928103f9d2642b034f4b6e8474d487d656fR174)